lib/src/form_builder_field.dart

```diff
@@ -88,7 +88,7 @@ class FormBuilderFieldState<F extends FormBuilderField<T>, T>
       (_formBuilderState?.initialValue ??
           const <String, dynamic>{})[widget.name] as T?;
 
-  dynamic get transformedValue => widget.valueTransformer?.call(value) ?? value;
+  dynamic get transformedValue => widget.valueTransformer == null ? value : widget.valueTransformer!(value);
```

Comment:
Why change this? Isn't the same thing that have before

Reply:
The main problem this is causing is using valueTransformer in textfield for example an "age" field that is optional.
If someone starts typing an age and then erases it the value is an empty String. Here the value transformer above should return null but instead you get the empty String as a return. The transformer is being run but because it returned null its value is discarded and instead you get "" as a return.